src/main/java/List.java

```diff
@@ -0,0 +1,61 @@
+public class List {
+    private Task[] tasks;
+    private int size;
+
+    public List() {
+        this.tasks = new Task[100]; // initial capacity
+        this.size = 0;
+    }
+
+    // Overloaded Method to add a deadline to the list
+    public void addTask(String taskDescription, String taskDeadline) {
+        tasks[size] = new Deadline(taskDescription, taskDeadline);
+        size++;
+    }
+
+    // Overloaded Method to add an event to the list
+    public void addTask(String taskDescription, String taskStart, String taskEnd) {
+        tasks[size] = new Event(taskDescription,taskStart, taskEnd);
+        size++;
+    }
+
+    // Method to add a task to the list
+    public void addTask(String taskDescription) {
+        tasks[size] = new ToDo(taskDescription);
+        size++;
+    }
+
+    // Method to print the last added task
+    public void printLastTask() {
+        if (size > 0) {
+            System.out.println("Last added task: " + tasks[size - 1]);
+        } else {
+            System.out.println("The list is empty.");
+        }
+    }
+
+    //Method to toggle check task as done by index
+    public void markDoneByIndex(int index){
+        tasks[index-1].completeTask();
+    }
+
+    //Method to toggle check task as undone by index
+    public void markUndoneByIndex(int index){
```

Comment:
Slight nitpick from me. Although egyptian style brackets were used throughout, having a space between the function parameters and curly brace could make the code look even neater.